From: Hugo Lefeuvre <hugo.lefeuvre@neclab.eu>
Subject: Add Mimalloc entry to the menuconfig and initialize it
diff --git a/lib/ukboot/Config.uk b/lib/ukboot/Config.uk
index 07139e2b..cfc306d7 100644
--- a/lib/ukboot/Config.uk
+++ b/lib/ukboot/Config.uk
@@ -32,6 +32,10 @@ if LIBUKBOOT
 		  Satisfy allocation as fast as possible. No support for free().
 		  Refer to help in ukallocregion for more information.
 
+		config LIBUKBOOT_INITMIMALLOC
+		bool "Mimalloc"
+		select LIBMIMALLOC
+
 		config LIBUKBOOT_NOALLOC
 		bool "None"
 
diff --git a/lib/ukboot/boot.c b/lib/ukboot/boot.c
index 4e749aa5..a400749d 100644
--- a/lib/ukboot/boot.c
+++ b/lib/ukboot/boot.c
@@ -45,6 +45,8 @@
 #include <uk/allocbbuddy.h>
 #elif CONFIG_LIBUKBOOT_INITREGION
 #include <uk/allocregion.h>
+#elif CONFIG_LIBUKBOOT_INITMIMALLOC
+#include <uk/mimalloc.h>
 #endif
 #if CONFIG_LIBUKSCHED
 #include <uk/sched.h>
@@ -233,6 +235,8 @@ void ukplat_entry(int argc, char *argv[])
 			a = uk_allocbbuddy_init(md.base, md.len);
 #elif CONFIG_LIBUKBOOT_INITREGION
 			a = uk_allocregion_init(md.base, md.len);
+#elif CONFIG_LIBUKBOOT_INITMIMALLOC
+			a = uk_mimalloc_init(md.base, md.len);
 #endif
 		} else {
 			uk_alloc_addmem(a, md.base, md.len);
